portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -925,7 +925,7 @@ else if (allowedDomain.equals(domain)) {
 				}
 				else if (allowedDomain.equals("PORTAL_DOMAINS")) {
 					if (Validator.isNotNull(
-							CompanyLocalServiceUtil.fetchCompanyByVirtualHost(
+							VirtualHostLocalServiceUtil.fetchVirtualHost(
```

Comment:
changed from `CompanyLocalServiceUtil.fetchCompanyByVirtualHost()` to `VirtualHostLocalServiceUtil.fetchVirtualHost()` which makes way more sense and will catch any virtual host on the current portal configurations, being it site or instance configuration. :)